sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -929,81 +929,95 @@ def sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns):
     return sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns)
 
 
-def df_getitem_str_literal_idx_main_codelines(self, idx):
+def df_getitem_slice_idx_main_codelines(self, idx):
     """Generate main code lines for df.getitem"""
-    try:
-        col_idx = self.columns.index(idx)
-    except ValueError:
-        func_lines = ['  raise KeyError']
+    if isinstance(self.index, types.NoneType):
+        func_lines = ['  length = len(get_dataframe_data(self, 0))',
+                      '  index = numpy.arange(length)']
     else:
-        col_data = f'data_{col_idx}'
-        func_lines = [
-            f'  {col_data} = get_dataframe_data(self, {col_idx})',
-            f'  return pandas.Series({col_data}, index=self._index, name=idx)'
+        func_lines = ['  index = self._index']
+
+    results = []
+    for i, col in enumerate(self.columns):
+        res_data = f'result_data_{i}'
+        func_lines += [
+            f'  data_{i} = get_dataframe_data(self, {i})',
+            f'  {res_data} = pandas.Series(data_{i}[idx], index=index[idx], name="{col}")'
         ]
+        results.append((col, res_data))
+
+    data = ', '.join(f'"{col}": {data}' for col, data in results)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}, index=index[idx])']
 
     return func_lines
 
 
-def df_getitem_str_literal_idx_codegen(self, idx):
+def df_getitem_str_slice_codegen(self, idx):
     """
-        def _df_getitem_str_literal_idx_impl(self, idx):
-          data_0 = get_dataframe_data(self, 0)
-          return pandas.Series(data_0, index=self._index, name=idx)
+    def _df_getitem_slice_idx_impl(self, idx):
+      index = self._index
```

Comment:
This comment is a bit incorrect, since index is defined this way only if it's not None.
```suggestion
      index = ...
```

Otherwise we might add overload for DataFrame.index similar to Series.index and use it in a generated func.

Reply:
The idea was to provide just an example of generated implementation in docstring. So I will update the docstring to say about that.
I don't mind the fact that we might add such overload, but I vote for doing it in separate PR.

Reply:
@kozlov-alexey what do you think about that?

Reply:
@densmirn I don't mind, let's do it in a separate one.